modules/nf-core/controlfreec/freec/tests/main.nf.test

```diff
@@ -0,0 +1,116 @@
+                input[3] = []
+                input[4] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz'], checkIfExists: true)
+                input[5] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz_tbi'], checkIfExists: true)
+                input[6] = UNTAR.out.untar.map{ it[1] }
+                input[7] = []
+                input[8] = file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
+                input[9] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.bedgraph,
+                    process.out.control_cpn,
+                    process.out.sample_cpn,
+                    process.out.gcprofile_cpn,
+                    process.out.BAF,
+                    process.out.CNV,
+                    process.out.info,
+                    process.out.ratio,
+                    process.out.versions).match() },
+                 { assert process.out.config }
+            )
+        }
+
+    }
+
+    test("controlfreec - single") {
+        options "-stub-run"
+        setup {
+            run("UNTAR"){
+                script "../../../untar/main.nf"
+                process {
+                    """
+                    input[0] = [ [id:'archive'],
+                                file(params.test_data['homo_sapiens']['genome']['genome_21_chromosomes_dir'], checkIfExists: true)
+                            ]
+                    """
+                }
+            }
+        }
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test2', single_end:false, sex:'XX' ], // meta map
+                    [],
+                    file(params.test_data['homo_sapiens']['illumina']['test2_mpileup'], checkIfExists: true),
+                    [],[],[],[]
+                ]
+
+                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
+                input[2] = file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
+                input[3] = []
+                input[4] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz'], checkIfExists: true)
+                input[5] = file(params.test_data['homo_sapiens']['genome']['dbsnp_138_hg38_21_vcf_gz_tbi'], checkIfExists: true)
+                input[6] = UNTAR.out.untar.map{ it[1] }
+                input[7] = []
+                input[8] = file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
+                input[9] = []
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success }
+            )
```

Comment:
need snap assertions here

Reply:
completely forgot about this PR. I am running stub tests here so all the files will be empty. _I think_ this is why I thought I don't need the assertion. What is the best practice here?

Reply:
The tools template is `{ assert snapshot(process.out).match() }`